urwid/widget/columns.py

```diff
@@ -691,12 +692,19 @@ def column_widths(self, size: tuple[int] | tuple[int, int], focus: bool = False)
             if t == WHSettings.GIVEN:
                 static_w = width
             elif t == WHSettings.PACK:
-                w_sizing = w.sizing()
-                if Sizing.FLOW in w_sizing:
+                if isinstance(w, Widget):
+                    w_sizing = w.sizing()
+                else:
+                    warnings.warn(f"{w!r} is not a Widget", ColumnsWarning, stacklevel=3)
+                    w_sizing = frozenset((urwid.BOX, urwid.FLOW))
+
+                if Sizing.FIXED in w_sizing:
+                    w_size = ()
+
+                elif Sizing.FLOW in w_sizing:
                     # FIXME: should be able to pack with a different maxcol value
                     w_size = (maxcol,)
```

Comment:
here still present part of bug: when we do not know widget size, we assume whole `cols` and element can be skipped.
Unfortunately, I do not see any good to make universal fix.